### PR TITLE
Origin/feature gateway

### PR DESCRIPTION
**Pull request by HALOXIAO**
init backend

### CODE REVIEW

**Comment by HALOXIAO** [comment]
framework：guice、vert.x

File: back-end/pom.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.function.stream</groupId>
```
Comment (RobertIndie) [inline back-end/pom.xml:7]:
```suggestion
    <groupId>org.functionstream</groupId>
```

File: back-end/gateway/pom.xml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>gateway</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+
+    <parent>
+        <artifactId>function-stream</artifactId>
+        <groupId>org.function.stream</groupId>
```
Comment (RobertIndie) [inline back-end/gateway/pom.xml:12]:
```suggestion
        <groupId>org.functionstream</groupId>
```

**Review by RobertIndie — CHANGES_REQUESTED** [review]
I saw a few empty interface classes. Is this PR not yet available for review?

File: back-end/gateway/src/main/java/org/functionstream/gateway/common/BusAddressConst.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.functionstream.gateway.common;
+
+/**
+ * @author HALOXIAO
```
Comment (RobertIndie) [inline back-end/gateway/src/main/java/org/functionstream/gateway/common/BusAddressConst.java:4]: According to 'Code Complete', comments are used to illustrate the purpose of the code. Using it for other purposes like specifying the ownership may result in 'comment decay'.
That said, keeping track of code ownership, change-log and who last modify the file, is the job of a source control tool like git, and should not be inside the comments.

I think we need to add some comments here to Illustrate what this interface is for.

File: back-end/gateway/pom.xml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>gateway</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+
+    <parent>
+        <artifactId>functionstream</artifactId>
```
Comment (RobertIndie) [inline back-end/gateway/pom.xml:11]:
```suggestion
        <artifactId>gateway</artifactId>
```

File: back-end/gateway/src/main/java/org/functionstream/gateway/service/ServerService.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Singleton;
+
+/**
+ * @author HALOXIAO
+ * @since 2021/7/21
+ */
+@Singleton
+public class ServerService {
```
Comment (RobertIndie) [inline back-end/gateway/src/main/java/org/functionstream/gateway/service/ServerService.java:11]: What is this ServerService class used for?

**Review by RobertIndie — CHANGES_REQUESTED** [review]

**Comment by RobertIndie** [comment]
Is there any update on this PR? @HALOXIAO

**Comment by HALOXIAO** [comment]
Done

**Comment by RobertIndie** [comment]
It looks like there are still a lot of comments left to fix. @HALOXIAO

**Pull request closed without merging**